e2e/aks_model.go

```diff
@@ -141,13 +146,52 @@ func airGapSecurityGroup(location, clusterFQDN string) (armnetwork.SecurityGroup
 	}, nil
 }
 
+func addPrivateEndpointToACR(ctx context.Context, nodeResourceGroup string, subnetId string) error {
+	endpointName := "PE-for-ABE2E"
+	peParams := armnetwork.PrivateEndpoint{
+		Location: to.Ptr(config.Config.Location),
+		Properties: &armnetwork.PrivateEndpointProperties{
+			Subnet: &armnetwork.Subnet{
+				ID: to.Ptr(subnetId),
+			},
+			PrivateLinkServiceConnections: []*armnetwork.PrivateLinkServiceConnection{
+				{
+					Name: to.Ptr(endpointName),
+					Properties: &armnetwork.PrivateLinkServiceConnectionProperties{
+						PrivateLinkServiceID: to.Ptr("/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.ContainerRegistry/registries/aksvhdtestcr"),
```

Comment:
did we make sure to delete-locks the new ACR?

Reply:
yep! I've had to remove it during testing since it affects the deletion of test private end points, but the plan is for there to be a delete lock.

Reply:
wait, how will the lock be removed when we create a new AKS cluster (hash changes?)